### PR TITLE
[POST-REBASE] remove hardcoded kubectl in apply warn msg

### PR DESCRIPTION
**Pull request by juanvallejo**
When a resource is passed to apply and it was not created using apply (or `oc create` with the `--save-config` flag), a warning message is printed: `Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply`.

This patch makes `kubectl` configurable in the warning message,

cc @stevekuznetsov @openshift/cli-review

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ func NewKubectlCommand(f cmdutil.Factory, in io.Reader, out, err io.Writer) *cob
 		{
 			Message: "Advanced Commands:",
 			Commands: []*cobra.Command{
-				NewCmdApply(f, out, err),
+				NewCmdApply("kubectl", f, out, err),
```
Comment (stevekuznetsov) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go]: Is there not a better source of truth for this?

Comment (juanvallejo) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go]: There doesn't seem to be one, in this file at least; closest thing would be to read the cobra command's `Use` string, which just seems to have "kubectl" [hardcoded in there](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go#L238) as well.

Best thing would probably be to have [NewKubectlCommand](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/kubectl/cmd/cmd.go#L235) receive a new argument `baseName string` and go from there.

Ultimate source of truth seems to [lie here](https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cli.go#L289-L294) :)

**Comment by juanvallejo** [comment]
[Upstream pr](https://github.com/kubernetes/kubernetes/pull/48894) has merged. @fabianofranz will just tag this one `post-rebase`

**Comment by liggitt** [comment]
message is configurable now:

`warningNoLastAppliedConfigAnnotation = "Warning: %[1]s apply should be used on resource created by either %[1]s create --save-config or %[1]s apply\n"`

**Pull request closed without merging**